### PR TITLE
refactor/docs

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- doc site example runner stinks. crashes all the time!

# solution

- move it to a webworker. 🐎 
- add more resilient error handling. self.onuncaughtrejection/self.onerror actually _work_ now 🤕 
- worry less about patching _normal_ window.console, because we're only patching the workers to send messages _back to the parent_, which is untouched 😌 
... annnnnd add an example that proves that current effect roots actually _dooo_ work. gotta better understand why that's the case.  expected it to break!

### CODE REVIEW

File: packages/effects-runtime/test/__fixtures__/functional-tests/concurrent-async-identity.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+function main () {
+  const aP = work('a')
+  const bP = work('b')
+  const cP = work('c')
+  return Promise.all([aP, bP, cP])
+}
+
+module.exports.test = ({it, expect}) => {
```
Comment (dpchamps) [inline packages/effects-runtime/test/__fixtures__/functional-tests/concurrent-async-identity.js:24]: Great test. Reminds me, i wrote some interrsting tests around competing concurrent actions at some point and I guess never committed them 🤔

File: packages/effects-docs/src/components/editor.worker.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+const effectsRuntimeP = import("effects-runtime/lib/prelude-polyfill.js");
+const transformEffectsP = import("babel-plugin-effects");
+import debounce from "lodash/debounce";
+
+// evil, mutable state
+let isEvaluating = false;
+let nextToEvaluate = "";
+
+async function resolveBabel() {
+  while (!global.Babel) await new Promise(res => setTimeout(res, 50));
```
Comment (dpchamps) [inline packages/effects-docs/src/components/editor.worker.js:19]: What if for some reason babel never loads?

Comment (cdaringe) [reply packages/effects-docs/src/components/editor.worker.js:19]: same if `/index.bundle.SHA.js` never loads. you get a network error and things just look broken. i opted to not design UX around this, as it involved message passing, new ui, etc.

**Comment by dpchamps** [comment]
🚢

**Review by dpchamps — APPROVED** [review]

**Pull request merged**